src/NuGet.Core/NuGet.Packaging/Signing/Content/SignatureManifest.cs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using NuGet.Versioning;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// Manifest stored in a signature containing the hash of a content manifest.
+    /// </summary>
+    public sealed class SignatureManifest
+    {
+        private const int MaxSize = 1024 * 1024;
```

Comment:
Why not use the MaxSize from `KeyPairFileUtility` ?

Reply:
It could, but these are very different files. This one should only have 3-5 lines so I left it separate to lower it later.